src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.Impl.cs

```diff
@@ -180,13 +180,13 @@ private void OnSymbolActionCore(ISymbol symbol, Action<Diagnostic> reportDiagnos
                 if (symbol.Kind == SymbolKind.NamedType)
                 {
                     var namedType = (INamedTypeSymbol)symbol;
-                    if (namedType.InstanceConstructors.Length == 1 &&
-                        (namedType.TypeKind == TypeKind.Class || namedType.TypeKind == TypeKind.Struct))
+                    if ((namedType.InstanceConstructors.Length == 1 && namedType.TypeKind == TypeKind.Class)
+                        || namedType.TypeKind == TypeKind.Struct)
                     {
-                        var instanceConstructor = namedType.InstanceConstructors[0];
-                        if (instanceConstructor.IsImplicitlyDeclared)
+                        var implicitConstructor = namedType.InstanceConstructors.SingleOrDefault(x => x.IsImplicitlyDeclared);
```

Comment:
My understanding is that we want to avoid as much as possible to avoid ending-up in situations where an analyzer throws so it might be better to either do a `FirstOrDefault` and disregard the case where we have multiple entries or do a `Where` and bail-out if we have too many items.

Reply:
I'd wait for @sharwell on this as this PR just a mirror on his PR in the old repo.

Reply:
+1 - It is almost always wrong to use `Single` or `SingleOrDefault` in analyzers/fixers when attempting to find matching symbols. My preference would be to handle this gracefully as suggested by @Evangelink

Reply:
```suggestion
                        var implicitConstructor = namedType.InstanceConstructors.FirstOrDefault(x => x.IsImplicitlyDeclared);
```